packages/flame/lib/src/components/component.dart

```diff
@@ -44,6 +44,17 @@ class Component {
   Component? get parent => _parent;
   Component? _parent;
 
+  /// Returns the first child that matches the given type [T].
+  T? child<T extends Component>() {
+    final match = children.whereType<T>();
+
+    if (match.isNotEmpty) {
+      return match.first;
+    }
+
+    return match.first;
```

Comment:
Shouldn't this be `return null`?

Anyways, a more efficient implementation would be
```dart
final it = match.iterator;
return it.moveNext()? it.current : null;
```
(this avoids running the iterator twice).